docs: activating ready live-examples

PedroOndh opened the pull request:
<!--Please provide a general summary of changes in the PR title -->
# Pull request template
<!--To help reviewers to understand the change you've made, you need to include **key information** in your PR.-->
Hello✖️  team! I prepared some **Live examples** through various components that are working right now in the eDocs portal. 

I would divide them in:
- Examples activated without further changes needed:
  - [ ] NextQueriesList
  - [ ] QuerySuggestion

- Examples where proposed some context changes (to match new API, juguettos --> Platform API, for example):
  - [ ] QuerySuggestions
  - [ ] RelatedTags
  - [ ] RelatedTag

- Examples where added the `<script>` tag to match the new format of the examples, included imports, `SearchInput` to test it out, basic data, etc. (but didn't need further changes)
  - [ ] HistoryQueries
  - [ ] HistoryQuery
  - [ ] ClearHistoryQueries
  - [ ] NextQueries
  - [ ] NextQuery
  - [ ] PopularSearches
  - [ ] PopularSearch
 
- Examples where proposed important changes to the given example
  - [ ] SearchInput
  - [ ] SearchButton
  - [ ] ClearSearchInput

## Motivation and context
<!--Include information on the purpose of the change and any background information that may help. Why is this change required? What problem does it solve? -->
This is created as a start point facilitated by eDocs to start activating again **Live examples** in the components documentation (and search for improvements by our own).

Our idea is to start delegating this effort in X team, giving you control to test the examples via [X Sandbox](https://docs-dev.empathy.co/x-sandbox/), and to open tasks for eDocs if you see improvements or errors in the tool.

A caveat right now, animations are registered globally in our portal (`FadeAndSlide`, `StaggeredFadeAndSlide`...), when doing this, we are doing an 'abstraction':

```
import Vue from 'vue';
...
import { FadeAndSlide } from '@empathyco/x-components';

// Registering the animation as a global component
Vue.component('FadeAndSlide', FadeAndSlide);
```

We are working in a way to do this more directly and we are adding [some documentation](https://docs-dev.empathy.co/x-sandbox/x-sandbox-documentation.html) ("Animations" section) refering to this. Ideas or recommendations are welcomed. 😄 
 
<!-- List any dependencies that are required for this change. If the change fixes an **open** issue, please link to the issue here.-->
- [x] Dependencies. If any, specify: **Technical Writters**, checking that "text changes" are correct
## Type of change
<!-- Indicate the type of change involved in the PR -->
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Change requires a documentation update
## What is the destination branch of this PR?
<!-- Although this may seem obvious, please include the destination branch as an extra check to ensure your PR targets the right branch.-->
- [x] `Main`
- [ ] Other. Specify:
## How has this been tested?
<!-- Please describe in detail how you tested your changes. Include details of your testing environment, the test cases used, and the tests you ran to see how your change affects other areas of the code, etc.-->
Applying this same changes to the eDocs project and checking that no errors were raised
## Checklist:
- [x] My code follows the **[style guidelines](./CONTRIBUTING.md#style-guides)** of this project.
- [x] I have performed a **self-review** on my own code.
- [x] I have **commented** my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation: ([open PR](https://github.com/empathyco/docs-empathy/pull/510/files) in eDocs with new documentation, [doc page](https://docs-dev.empathy.co/x-sandbox/x-sandbox-documentation.html))
- [x] My changes generate **no new warnings**.
- [ ] I have added **tests** that prove my fix is effective or that my feature works.
- [x] New and existing **unit tests pass locally** with my changes.

CachedaCodes commented on packages/x-components/src/x-modules/history-queries/components/clear-history-queries.vue:
````diff
@@ -83,10 +83,30 @@
-  <img class="x-history-query__icon" src="./my-icon.svg"/>
-</ClearHistoryQueries>
+```vue live
+<template>
````
`<template>` can only have one root child. Wrap the components in a `div` for the code to work

CachedaCodes commented on packages/x-components/src/x-modules/history-queries/components/history-queries.vue:
````diff
@@ -104,15 +104,51 @@ This component renders a list of suggestions coming from the user queries histor
-```vue
-<HistoryQueries />
+```vue live
+<template>
````
`<template>` can only have one root child. Wrap the components in a div for the code to work

CachedaCodes commented on packages/x-components/src/x-modules/history-queries/components/history-queries.vue:
````diff
@@ -104,15 +104,51 @@ This component renders a list of suggestions coming from the user queries histor
-```vue
-<HistoryQueries :animation="FadeAndSlide" :maxItemsToRender="10" />
+```vue live
+<template>
````
`<template>` can only have one root child. Wrap the components in a `div` for the code to work

CachedaCodes commented on packages/x-components/src/x-modules/history-queries/components/history-queries.vue:
````diff
@@ -122,13 +158,30 @@ The default `HistoryQuery` component that is used in every suggestion can be rep
-  </template>
-</HistoryQueries>
+```vue live
+<template>
````
`<template>` can only have one root child. Wrap the components in a div for the code to work

CachedaCodes commented on packages/x-components/src/x-modules/history-queries/components/history-queries.vue:
````diff
@@ -137,15 +190,31 @@ The content of the `HistoryQuery` component can be overridden. For replacing the
-  </template>
-</HistoryQueries>
+```vue live
+<template>
````
`<template>` can only have one root child. Wrap the components in a div for the code to work

CachedaCodes commented on packages/x-components/src/x-modules/history-queries/components/history-query.vue:
````diff
@@ -91,8 +91,32 @@
-```vue
-<HistoryQuery :suggestion="historyQuery" />
+```vue live
+<template>
````
`<template>` can only have one root child. Wrap the components in a div for the code to work

CachedaCodes commented on packages/x-components/src/x-modules/history-queries/components/history-query.vue:
`````diff
@@ -105,18 +129,47 @@ the suggestion itself, and a `string` of HTML with the matched query.
-event is emitted after the user clicks the button. The event payload is the history query data.
-````
+```vue live
+<template>
`````
`<template>` can only have one root child. Wrap the components in a div for the code to work

CachedaCodes commented on packages/x-components/src/x-modules/next-queries/components/next-queries.vue:
````diff
@@ -100,16 +100,52 @@
-```vue
-<NextQueries :animation="FadeAndSlide" :maxItemsToRender="10" :highlightCurated="true" />
+```vue live
+<template>
````
`<template>` can only have one root child. Wrap the components in a div for the code to work

CachedaCodes commented on packages/x-components/src/x-modules/next-queries/components/next-queries.vue:
````diff
@@ -118,13 +154,29 @@ You can use your custom implementation of the Next Query's content. In the examp
-  </template>
-</NextQueries>
+```vue live
+<template>
````
`<template>` can only have one root child. Wrap the components in a div for the code to work

CachedaCodes commented on packages/x-components/src/x-modules/next-queries/components/next-queries.vue:
````diff
@@ -134,17 +186,34 @@ the `emitNextQuerySelected` function when the next query is selected. In the exa
-  </template>
-</NextQueries>
+```vue live
+<template>
````
`<template>` can only have one root child. Wrap the components in a div for the code to work

CachedaCodes commented on packages/x-components/src/x-modules/search-box/components/clear-search-input.vue:
````diff
@@ -124,11 +126,10 @@ In this example, the `UserPressedClearSearchBoxButton` event is implemented, tri
 
-```vue
+```vue live
 <template>
````
`<template>` can only have one root child. Wrap the components in a div for the code to work

CachedaCodes commented on packages/x-components/src/x-modules/search-box/components/search-button.vue:
````diff
@@ -145,24 +145,34 @@ search term and click the Search button, the search term is displayed as a messa
 
-```vue
+```vue live
 <template>
````
`<template>` can only have one root child. Wrap the components in a div for the code to work

CachedaCodes commented on packages/x-components/src/x-modules/search-box/components/search-button.vue:
````diff
@@ -176,10 +186,13 @@ you enter a search term and click the Search button, the “Looking for results
 
-```vue
+```vue live
 <template>
````
`<template>` can only have one root child. Wrap the components in a div for the code to work

CachedaCodes commented on packages/x-components/src/x-modules/search-box/components/search-input.vue:
````diff
@@ -319,14 +319,14 @@ the message “enter” appears.
 
-```vue
+```vue live
 <template>
````
`<template>` can only have one root child. Wrap the components in a div for the code to work

CachedaCodes commented on packages/x-components/src/x-modules/next-queries/components/next-queries.vue:
````diff
@@ -100,16 +100,52 @@
-```vue
-<NextQueries />
+```vue live
+<template>
````
`<template>` can only have one root child. Wrap the components in a div for the code to work

PedroOndh reviewed (commented):
Oh, you are right! 

As our "[sandbox](https://docs-dev.empathy.co/x-sandbox/)" already add a root element, and made it work, i didn't realized we are giving a bad advice to doc readers. 😅 

I solved these cases (in almost every case wrapping them in a `div`) and checked that styles are still working as they should.

PedroOndh commented:
I rebased this branch with `main` to solve conflicts and changed some details in `history-queries`, most important one is changing `#suggestion-content-remove` to `#suggestion-remove-content`, as it didn't work the other way.

tiborux commented on packages/x-components/src/x-modules/history-queries/components/history-queries.vue:
```diff
@@ -50,6 +50,7 @@
   import BaseSuggestions from '../../../components/suggestions/base-suggestions.vue';
   import { Getter } from '../../../components/decorators/store.decorators';
   import { xComponentMixin } from '../../../components/x-component.mixin';
+  import SearchInput from '../../search-box/components/search-input.vue';
```
Remove this

PedroOndh replied:
Wow! You are right, i think (hope) that IDE added this. I am removing it.

tiborux commented on packages/x-components/src/x-modules/history-queries/components/history-queries.vue:
```diff
@@ -65,7 +66,7 @@
    */
   @Component({
-    components: { BaseSuggestions, HistoryQuery },
+    components: { SearchInput, BaseSuggestions, HistoryQuery },
```
Remove this

PedroOndh replied:
Wow! You are right, i think (hope) that IDE added this. I am removing it.

tiborux commented on packages/x-components/src/x-modules/history-queries/components/history-query.vue:
```diff
@@ -91,8 +91,31 @@
+</template>
+
+<script>
+  import { SearchInput } from '@empathyco/x-components/search-box';
```
Remove this then

tiborux commented on packages/x-components/src/x-modules/history-queries/components/history-query.vue:
```diff
@@ -91,8 +91,31 @@
+  export default {
+    name: 'HistoryQueryDemo',
+    components: {
+      SearchInput,
```
And this line

PedroOndh replied:
Removed

tiborux commented on packages/x-components/src/x-modules/history-queries/components/history-query.vue:
```diff
@@ -105,18 +128,46 @@ the suggestion itself, and a `string` of HTML with the matched query.
+</template>
+
+<script>
+  import { SearchInput } from '@empathyco/x-components/search-box';
```
```suggestion
```

PedroOndh replied:
Removed

tiborux commented on packages/x-components/src/x-modules/history-queries/components/history-query.vue:
```diff
@@ -105,18 +128,46 @@ the suggestion itself, and a `string` of HTML with the matched query.
+  export default {
+    name: 'HistoryQueryDemo',
+    components: {
+      SearchInput,
```
```suggestion
```

PedroOndh replied:
Removed

tiborux commented on packages/x-components/src/x-modules/related-tags/components/related-tag.vue:
```diff
@@ -218,10 +218,10 @@ _See how the related tag can be rendered._
       return {
         tag: {
           modelName: 'RelatedTag',
```
Use our model:
  
```
  tag: {
          modelName: 'RelatedTag',
          query: 'high heel',
          isCurated: false,
          tag: 'heel'
        }
```

PedroOndh replied:
Great! I changed it. :D

tiborux commented on packages/x-components/src/x-modules/related-tags/components/related-tag.vue:
```diff
@@ -253,10 +253,10 @@ _See how the event is triggered when the related tag is clicked._
       return {
         tag: {
```
Use our model:
  
```
  tag: {
          modelName: 'RelatedTag',
          query: 'high heel',
          isCurated: false,
          tag: 'heel'
        }
```

tiborux commented on packages/x-components/src/x-modules/related-tags/components/related-tag.vue:
```diff
@@ -182,10 +182,10 @@ _Here you can see how the RelatedTag component is rendered._
       return {
         tag: {
```
Use our model:
  
```
  tag: {
          modelName: 'RelatedTag',
          query: 'high heel',
          isCurated: false,
          tag: 'heel'
        }
```

PedroOndh replied:
Great! I changed it. :D

tiborux commented on packages/x-components/src/x-modules/search-box/components/search-input.vue:
```diff
@@ -319,14 +319,16 @@ the message “enter” appears.
-  />
+  <div>
+    <SearchInput
+      @UserPressedEnterKey="value = 'enter'"
```
Why you remove    @UserBlurredSearchBox="hasFocus = false"?

PedroOndh replied:
I did it because when pressing "Enter" `UserPressedEnterKey` and `UserBlurredSearchBox` were both called, so they were conflicting, but now that we are talking about this, i think that it's better to use value and hasFocus independently, so we can conserve the event and see them in real time

tiborux reviewed (changes requested):
<img src="https://media0.giphy.com/media/mGK1g88HZRa2FlKGbz/giphy.gif"/>

tiborux commented on packages/x-components/src/x-modules/history-queries/components/history-query.vue:
```diff
@@ -105,18 +126,44 @@ the suggestion itself, and a `string` of HTML with the matched query.
+<template>
+  <HistoryQuery :suggestion="suggestion">
+    <template #default="{ suggestion, queryHTML }">
+      <img class="x-history-query__history-icon" src="/assets/icons/bulb.svg" />
```
```suggestion
  <LightBulbOn />
```

tiborux commented on packages/x-components/src/x-modules/history-queries/components/history-query.vue:
```diff
@@ -105,18 +126,44 @@ the suggestion itself, and a `string` of HTML with the matched query.
+    </template>
+
+    <template #remove-button-content="{ suggestion }">
+      <img class="x-history-query__remove-icon" src="/assets/icons/cross-dark.svg" />
```
```suggestion
  <CrossIcon />
```

tiborux commented on packages/x-components/src/x-modules/history-queries/components/history-query.vue:
```diff
@@ -105,18 +126,44 @@ the suggestion itself, and a `string` of HTML with the matched query.
+  export default {
+    name: 'HistoryQueryDemo',
+    components: {
+      HistoryQuery
```
```suggestion
      HistoryQuery,
    CrossIcon,
    LightBulbOn
```

tiborux commented on packages/x-components/src/x-modules/history-queries/components/history-query.vue:
```diff
@@ -105,18 +126,44 @@ the suggestion itself, and a `string` of HTML with the matched query.
+</template>
+
+<script>
+  import { HistoryQuery } from '@empathyco/x-components/history-queries';
```
```suggestion
  import { HistoryQuery } from '@empathyco/x-components/history-queries';
  import { CrossIcon, LightBulbOn } from '@empathyco/x-components';
```

tiborux commented on packages/x-components/src/x-modules/next-queries/components/next-queries.vue:
```diff
@@ -118,13 +158,31 @@ You can use your custom implementation of the Next Query's content. In the examp
+    <SearchInput />
+    <NextQueries>
+      <template #suggestion-content="{ suggestion }">
+        <img src="/assets/icons/bulb.svg" class="x-next-query__icon" />
```
Change this icon for the BulbOn, the one from the previous comment.

PedroOndh replied:
I am trying to add more specific icons, matching this component, i added `Nq1Icon`.

tiborux commented on packages/x-components/src/x-modules/next-queries/components/next-queries.vue:
```diff
@@ -134,17 +192,36 @@ the `emitNextQuerySelected` function when the next query is selected. In the exa
+      <template #suggestion="{ suggestion }">
+        <NextQuery :suggestion="suggestion" class="x-next-queries__suggestion">
+          <template #default="{ suggestion }">
+            <img src="/assets/icons/bulb.svg" class="x-next-query__icon" />
```
Change this icon for the BulbOn, the one from the previous comment.

PedroOndh replied:
I am trying to add more specific icons, matching this component, i added `Nq1Icon`.

tiborux commented on packages/x-components/src/x-modules/popular-searches/components/popular-search.vue:
```diff
@@ -67,21 +67,63 @@ content. By default, it renders the suggestion query of the popular search.
+<template>
+  <PopularSearch :suggestion="suggestion">
+    <template #default="{ suggestion }">
+      <svg height="10" width="10">
```
Change this svg by `TrendingIcon` that its inside x-components

tiborux commented on packages/x-components/src/x-modules/popular-searches/components/popular-searches.vue:
```diff
@@ -102,13 +132,25 @@ You can use your custom implementation of the Popular Search's content. In the e
+<template>
+  <PopularSearches>
+    <template #suggestion-content="{ suggestion }">
+      <img class="x-popular-search__icon" src="/assets/icons/bulb.svg" />
```
Change this icon for the bulb and import it.

PedroOndh replied:
I am trying to add more specific icons, matching this component, i added `TrendingIcon`.

tiborux commented on packages/x-components/src/x-modules/related-tags/components/related-tags.vue:
````diff
@@ -112,9 +112,9 @@ To use this component, the QuerySignals microservice must be implemented.
+```vue live
 <template>
   <div>
     <SearchInput></SearchInput>
````
I just found this, all the components that are not using the slots the have to be self-closed: https://v2.vuejs.org/v2/style-guide/?redirect=true#Self-closing-components-strongly-recommended

So please, check all the files that you touched and change the components that can be self-closed to be self-closed.

PedroOndh replied:
I searched through the examples and changed to self-closed some components in `related-tags`, `related-tag` and `history-queries`.

tiborux commented on packages/x-components/src/x-modules/search-box/components/search-input.vue:
```diff
@@ -319,14 +319,18 @@ the message “enter” appears.
+      @UserIsTypingAQuery="value = 'typing'"
+    />
+    <strong>{{ value }}</strong>
+    {{ hasFocus ? 'focused' : 'unfocused' }}
```
```suggestion
   <span> {{ hasFocus ? 'focused' : 'unfocused' }} </span>
```

tiborux reviewed (changes requested):
Gj! just minor things

PedroOndh commented on packages/x-components/src/x-modules/next-queries/components/next-query.vue:
```diff
@@ -92,21 +92,65 @@ CSS class.
+<template>
+  <NextQuery :suggestion="suggestion">
+    <template #default="{ suggestion }">
+      <img class="x-next-query__icon" src="/assets/icons/bulb.svg" />
```
I am trying to add more specific icons, matching this component, i added `Nq1Icon`.

PedroOndh commented:
Made every change. 😄 

I didn't used `LightBulbOn` in the end, both because it's not included yet in the X components version we are using (44, we will have to dedicate a task to update it as it will have some complexity) and because i tried to show something more specific. This way we can use this examples also as a showcase of icons.

tiborux reviewed (approved):
<img src="https://media3.giphy.com/media/111ebonMs90YLu/giphy.gif"/>
